internal/cortex/frontend/transport/handler.go

```diff
@@ -63,12 +67,30 @@ type Handler struct {
 	activeUsers  *util.ActiveUsersCleanupService
 }
 
+// NormalizeString Condeneses whitespaces in string; does not remove all whitespaces
+func NormalizeString(inp string) string {
+	normalized := regexp.MustCompile(`[\n\t]+`).ReplaceAllString(inp, " ")
+	normalized = regexp.MustCompile(`\s+`).ReplaceAllString(normalized, " ")
```

Comment:
Move regexp.MustCompile(`[\n\t]+`) and regexp.MustCompile(`\s+`) into `type Handler struct` so that they can be shared across requests instead of compiling them every time.

<img width="1245" alt="image" src="https://github.com/databricks/thanos/assets/129140017/0c5c113f-2723-4c3d-9dec-e7c2f7467a04">